obspy/clients/nrl/client.py

```diff
@@ -0,0 +1,304 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+'''
+Class for navigating, and picking responses from the
+Nominal Response Library.
+
+:copyright:
+    Lloyd Carothers IRIS/PASSCAL, 2016
+:license:
+    GNU Lesser General Public License, Version 3
+    (http://www.gnu.org/copyleft/lesser.html)
+'''
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+
+import codecs
+import io
+import os
+import sys
+import requests
+
+from obspy.core.inventory.util import _textwrap
+from obspy.io.xseed import Parser
+
+if sys.version_info.major == 2:
+    import ConfigParser as configparser
+else:
+    import configparser
+
+
+class NRL(object):
+    """
+    NRL client base class for accessing the Nominal Response Library:
+    http://ds.iris.edu/NRL/
+
+    Created with a URL for remote access or filesystem accessing a local copy.
+    """
+
+    _index = 'index.txt'
+
+    def __new__(cls, root=None):
+        if root is not None:
+            o = requests.utils.urlparse(root)
+
+        if root is None or o.scheme == 'http':
```

Comment:
I'm easy, but am happy with it as so...
`if root is not None`  line 44 will run and o will exist.

Python short circuits so `if root is None`  o.scheme will not get hit